lib/src/view/broadcast/broadcast_game_screen.dart

```diff
@@ -0,0 +1,509 @@
+                          borderRadius: BorderRadius.circular(4.0),
+                        ),
+                        child: EngineGauge(
+                          displayMode: EngineGaugeDisplayMode.vertical,
+                          params: engineGaugeParams,
+                        ),
+                      );
+              }
+            : null,
+        engineLines: isLocalEvaluationEnabled && numEvalLines > 0
+            ? EngineLines(
+                clientEval: currentNode.eval,
+                isGameOver: currentNode.position.isGameOver,
+                onTapMove: ref
+                    .read(
+                      broadcastGameControllerProvider(roundId, gameId).notifier,
+                    )
+                    .onUserMove,
+              )
+            : null,
+        bottomBar: BroadcastGameBottomBar(roundId: roundId, gameId: gameId),
+        children: [BroadcastGameTreeView(roundId, gameId)],
+      ),
+    );
+  }
+}
+
+class _BroadcastBoardWithHeaders extends ConsumerWidget {
+  final BroadcastRoundId roundId;
+  final BroadcastGameId gameId;
+  final double size;
+  final Radius? radius;
+
+  const _BroadcastBoardWithHeaders(
+    this.roundId,
+    this.gameId,
+    this.size,
+    this.radius,
+  );
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    // TODO use borderRadius with players widget
+    return Column(
+      children: [
+        _PlayerWidget(
+          roundId: roundId,
+          gameId: gameId,
+          width: size,
+          widgetPosition: _PlayerWidgetPosition.top,
+          radius: radius ?? Radius.zero,
+        ),
+        _BroadcastBoard(roundId, gameId, size, radius != null),
+        _PlayerWidget(
+          roundId: roundId,
+          gameId: gameId,
+          width: size,
+          widgetPosition: _PlayerWidgetPosition.bottom,
+          radius: radius ?? Radius.zero,
+        ),
+      ],
+    );
+  }
+}
+
+class _BroadcastBoard extends ConsumerStatefulWidget {
+  const _BroadcastBoard(
+    this.roundId,
+    this.gameId,
+    this.boardSize,
+    this.hasRadius,
+  );
+
+  final BroadcastRoundId roundId;
+  final BroadcastGameId gameId;
+  final double boardSize;
+  final bool hasRadius;
+
+  @override
+  ConsumerState<_BroadcastBoard> createState() => _BroadcastBoardState();
+}
+
+class _BroadcastBoardState extends ConsumerState<_BroadcastBoard> {
+  ISet<Shape> userShapes = ISet();
+
+  @override
+  Widget build(BuildContext context) {
+    final ctrlProvider =
+        broadcastGameControllerProvider(widget.roundId, widget.gameId);
+    final broadcastAnalysisState = ref.watch(ctrlProvider).requireValue;
+    final boardPrefs = ref.watch(boardPreferencesProvider);
+    final showBestMoveArrow = ref.watch(
+      analysisPreferencesProvider.select(
+        (value) => value.showBestMoveArrow,
+      ),
+    );
+    final showAnnotationsOnBoard = ref.watch(
+      analysisPreferencesProvider.select((value) => value.showAnnotations),
```

Comment:
Instead of using 2 `.select` prefer, watching directly the whole analysis preferences state.

Reply:
Why do you think it is better to watch the whole analysis preferences state ? I'd like to understand so I don't do the same mistakes again.

From what I understand, by listening to the whole state it will rebuild the board even if an analysis preference shouldn't change the board.

Reply:
Because the code is less readable (several watch instead of one) and the "optimisation" is not worth it. Analysis prefs will rarely change and it is ok to rebuild even if one of the change is not related to this screen.